Makefile

```diff
@@ -202,6 +202,17 @@ THIRD_PARTY_NOTICE_LICENSE_TYPE = microsoft-oss
 $(eval $(call CREATE_THIRD_PARTY_NOTICES_RULE,ThirdPartyNotices.txt,foundation,False,False))
 $(eval $(call CREATE_THIRD_PARTY_NOTICES_RULE,bin/$(CONFIGURATION)/lib/xamarin.android/ThirdPartyNotices.txt,$(THIRD_PARTY_NOTICE_LICENSE_TYPE),True,False))
 
+EXTERNAL_GIT_PATH=$(topdir)/external
+XA_PATH=$(topdir)
+prepare-external-git-dependencies:
+	$(call MSBUILD_BINLOG,prep-tasks) build-tools/xa-prep-tasks/xa-prep-tasks.csproj
+	$(call MSBUILD_BINLOG,external-checkout) build-tools/xa-prep-tasks/xa-prep-tasks.csproj \
+		/t:CheckoutExternalGitSources /p:ExternalSourceDependencyDirectory='$(EXTERNAL_GIT_PATH)'
+	cd $(MONODROID_PATH) && ./configure --with-xamarin-android='$(XA_PATH)'
```

Comment:
Are we missing something?  I don't see a declaration/assignment of `$(MONODROID_PATH)`, so this will presumably expand to `cd && ./configure ...`, which would run `$HOME/configure`, which would be very unexpected.

I think this should be split up into two targets: `prepare-external` and `prepare-commercial`:

```makefile
prepare-external:
	$(call MSBUILD_BINLOG,prep-tasks) build-tools/xa-prep-tasks/xa-prep-tasks.csproj
	$(call MSBUILD_BINLOG,external-checkout) build-tools/xa-prep-tasks/xa-prep-tasks.csproj \
			/t:CheckoutExternalGitSources /p:ExternalSourceDependencyDirectory='$(EXTERNAL_GIT_PATH)'

prepare-commercial: prepare-external
	cd $(MONODROID_PATH) && ./configure --with-xamarin-android='$(XA_PATH)'
```

Reply:
This was very broken... I had only been testing through msbuild when making updates. This should now be resolved.